kaldb/src/main/java/com/slack/kaldb/metadata/zookeeper/ZookeeperMetadataStoreImpl.java

```diff
@@ -51,6 +54,20 @@
   public static final String METADATA_WRITE_COUNTER = "metadata.write";
   public static final String METADATA_READ_COUNTER = "metadata.read";
 
+  private static final int ZK_RETRY_COUNT = 3;
```

Comment:
This should probably be part of the ZkConfig

Reply:
I left in as a hard coded value intentionally. If we failed thrice for a session reconnect, we usually terminate the VM, so this is fine for now.